infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServerCache.java

```diff
@@ -90,7 +93,10 @@ public ILogData get(long address) {
      * @param entry   the log entry to write
      */
     public void put(long address, ILogData entry) {
-        log.trace("LogUnitServerCache.put: Cache write[{} : {}]", address, entry);
+        log.info("LogUnitServerCache.put: Cache write[{} : {}]", address, entry);
+        if (entry instanceof LogData  && !(streamLog instanceof InMemoryStreamLog)) {
+            ((LogData)entry).serializeMetadata();
+        }
```

Comment:
Should remove the checking of memory log.

Reply:
Removed